metabase/metabase
# Custom Table e2e Tests 1: Add basic querying for postgres and mysql

## Description

- uses [knex](https://knexjs.org/guide/raw.html) to allow raw queries to both postgres and mysql via the `queryQADB()` function
- adds preset configs for mysql and postgres DB configs to make the developer experience easier

## Usage

```js
import {
  queryQADB,
} from "__support__/e2e/helpers";

it("can select in mysql", () => {
  queryQADB('SELECT 123 as "foo"', 'mysql').then((result) => {
    expect(result.rows).to.deep.equal([{ foo: 123 }]);
  });
});

it("can select in postgres", () => {
  queryQADB('SELECT 123 as "foo"', 'postgres').then((result) => {
    expect(result.rows).to.deep.equal([{ foo: 123 }]);
  });
});

```

![Screen Shot 2023-01-31 at 2 26 28 PM](https://user-images.githubusercontent.com/30528226/215887791-266d9bb5-73ec-4a3b-b8d1-7a25e1ccb02f.png)

## How to test
- start cypress locally using `yarn test-cypress-open-qa`
- paste the block above into any old test and see that it works!
